Makefile

```
@@ -454,6 +454,37 @@ $(BIN)/tesseract: tesseract/configure
 	cd $(VIRTUAL_ENV)/build/tesseract && $(CURDIR)/tesseract/configure --disable-openmp --disable-shared --prefix="$(VIRTUAL_ENV)" CXXFLAGS="-g -O2 -fPIC"
 	cd $(VIRTUAL_ENV)/build/tesseract && $(MAKE) install
 
+# Build and install Tesseract training tools.
+
+TESSTRAIN_EXECUTABLES =
+TESSTRAIN_EXECUTABLES += $(BIN)/ambiguous_words
+TESSTRAIN_EXECUTABLES += $(BIN)/classifier_tester
+TESSTRAIN_EXECUTABLES += $(BIN)/cntraining
+TESSTRAIN_EXECUTABLES += $(BIN)/combine_lang_model
+TESSTRAIN_EXECUTABLES += $(BIN)/combine_tessdata
+TESSTRAIN_EXECUTABLES += $(BIN)/dawg2wordlist
+TESSTRAIN_EXECUTABLES += $(BIN)/lstmeval
+TESSTRAIN_EXECUTABLES += $(BIN)/lstmtraining
+TESSTRAIN_EXECUTABLES += $(BIN)/merge_unicharsets
+TESSTRAIN_EXECUTABLES += $(BIN)/mftraining
+TESSTRAIN_EXECUTABLES += $(BIN)/set_unicharset_properties
+TESSTRAIN_EXECUTABLES += $(BIN)/shapeclustering
+TESSTRAIN_EXECUTABLES += $(BIN)/text2image
+TESSTRAIN_EXECUTABLES += $(BIN)/unicharset_extractor
+TESSTRAIN_EXECUTABLES += $(BIN)/wordlist2dawg
+
+.PHONY: install-tesseract-training
+install-tesseract-training: $(TESSTRAIN_EXECUTABLES)
+
+$(TESSTRAIN_EXECUTABLES): tesseract/configure
```

This will ensure that the rule does not fire up N times for N tools, but make will treat the recipe as multi-target (without waiting for make 4.3's proper "grouped target" support):

```suggestion
$(call multirule,$(TESSTRAIN_EXECUTABLES)): tesseract/configure
```